### ISSUE TITLE
Potential heli crash site changes

### ISSUE THREAD

**Issue by bchawke** [DayZMod/DayZ#834]
Crash sites have been coming up in discussion a lot lately so I figured I'd open an issue to discuss what can be done to make them better.  Here's some of my suggestions (some of these based on other peoples' comments I've seen)

- Adjust spawn mechanic so that there are no crash sites at server restart.  Then carry on with the currently implemented mechanic of having random chances of them spawning over the course of the server cycle.  This would prevent heli squads from harvesting crash sites right after reset and reduce the gap between lone players and squads.
- Set a maximum number of crash sites that can be present on the map.  Three, maybe?  We've seen the hilarious images of servers that have not been reset having dozens of crash sites littered about.
- Assuming the above suggestions are taken into account and crash sites are more rare, perhaps some of the lower tiered weapons should be removed from their loot tables.  I find no need for Bizons, AK74s, or M16s to spawn there.  Players that *do* find the rare crash sites should have a higher chance of being rewarded with weapons that cannot be looted at "MilitaryLow" or "Military" class buildings.
- I love @M0tzilla's idea of having zombies perma-spawned on crash sites with a radius that they can wander.  Is this possible?  The downside would be that if there are no zombies you'd know that a player has looted it.  Perhaps have zombies respawn within 5 minutes so that zombies are present for the duration of the server cycle?  I doubt this would be a big hit to server performance with between 0 and 3 crash sites being present

How do people feel about the medical crash sites?  I certainly like the idea of them being in the game, however one aspect I miss about earlier versions of DayZ Mod is that you were forced to go to the coast if you ran out of medical supplies.  I miss being fully geared and having to risk a Cherno/Elektro/Berezino run to find antibiotics, morphine, or a blood back like we had to in the old days.  The coast had some real value when it was the only place to find med supplies.

**Comment by skigoggles** [comment]
Lots of issues here. All urban legends aside, ONE heli crash spawns at restart, then every 25 minutes with some variance, there is a 75% chance another will spawn. So, you are talking 1 vs. 0.

* MAX heli crashes are a function of the time before restart.  If the server never restarts, then cleanup of destroyed items, base maintenance and the like would not happen, so not sure why that's a reason to change anything.

* How many crash sites are there? My server restarts every 6 hours and doing a quick search through a few days logs I'd say the average is 9 per session.

* Not sure how suddenly this turned into making crash sites more rare

And it wasn't said here but in the other thread, that clans with helis get to all the crash sites - yup someone with a heli can fly all around the map quickly. The placement is random, thats been proven over and over again, so thats more of a statement that an issue.

**Comment by bchawke** [comment]
The 1 vs 0 is actually a big deal.  Any squad with a heli is practically *guaranteed* a clean heli crash site to loot at every server restart currently.  That's a huge advantage.  Squads should _definitely_ have an advantage, but not a huge advantage.  I've actually played some late nights and observed heli crews log on at restart, fly their heli around for an hour to harvest the 1-3 crash sites, then log off, rinse, repeat, like clockwork.

To be fair, it's not _just_ a matter of "make them more rare", but rather have them start spawning after restart rather than right at restart, and limit the total number to 3 or 4, but also spawn high end loot more consistently rather than spawning coastal SMGs and small caliber assault rifles.  Perhaps more heli crashes with a lower chance of top tier loot is more balanced.  Just throwing ideas out there, open to discussion.

**Comment by thcNox** [comment]
Vanilla is designed around a 3 hour server up time in which you will on average have 4-6 crash sites spawn randomly during every reset. Which is fine imo. Of course your position is to stop roaming in favor of re-establishing the NWAF meltingpot in which case it makes perfect sense to lower the impact of helicrashes and bloat the barracks loot table which I simply disagree with.

**Comment by bchawke** [comment]
Your assumptions about my position are incorrect, Nox.  My position is to narrow the enormous gap between new players/lone wolves and geared squads that the mod currently suffers from.  My main interest in that regard is having the server start without a heli crash spawned in at the beginning.  As I stated in my above comment, I'm on the fence as to whether more heli crashes with lower tiered loot or fewer heli crashes with higher tiered loot make for better balance and am open to discussion about it.  I'm a huge fan of heli crashes carrying a lot of value.  I think it's fantastic that the highest tier loot such as DMR/NVG/FAL/Mk48/etc spawn exclusively at heli crashes.  I've expressed a number of times that I'm a fan of the expansion of potential spawn points further N, E, and W which encourages people to venture further out there looking for crash sites.  It has nothing to do with "stopping roaming".  As I've stated dozens of times in various discussions, I'm in favor of giving people incentive to roam the map, but am of the opinion the neutering the barracks was the wrong approach to achieving that goal as it completely removed a popular strategic hot spot on the map that many players liked to engage in PvP at.  The ideal goal is to have NWAF be a popular hot spot _AND_ give players incentive to venture to other areas of the map.  It is possible to have both, which is what I'm advocating, _NOT_ to "stop roaming in favor of re-establishing the NWAF meltingpot".

**Comment by thcNox** [comment]
When I voiced my fears about NWAF returning to the state it had been in 1.8.6 and earlier patches in #833 (which prompted the gutting of the barracks loot table to begin with) You made it clear that you did not agree that there ever had been a problem but rather hailed it as "one of the best design features of DayZ Mod." And now here you with a new issue wanting to limit the impact of heli crashes making hilarious suggestions like effectively cutting the amount of crashes in half based upon pictures you have seen of somebody's test server that's only restarted manually or from some none vanilla server which is hardly relevant on this github. 

I don't know how one is supposed to draw any other conclusion from your voiced opinions and suggested changes that you wish to have focus return to the NWAF and diminish the viability and lure of roaming for crashsites. 

As for justifying these changes by claiming they will narrow the "enormous" gap between new players/lone wolfs and geared squads I don't see how that's supposed to happen. Generally people who on top of spending a lot of time playing the game also group up will always wreck people who don't. Gear has little impact on this relationship. Also with your changes to the barracks lone wolfers and new players will have a harder time looting airfields than before meanwhile geared squads will loot cycle at will further bloating the loot economy in favor of geared players.

**Comment by Streatman** [comment]
what @thcNox said

ppl already build bases as nwaf and camp there, and ppl already complain about that
lowering crashsides and raise barracks will only result in even more of that, u take the only way for a lonewolf to find good stuff and want to establish a system were squads are even more favored... did u ever played dayzero and do u know how it turned out?

if u have a problem with helicopter groups farming the map then u can regulate that by fuel rarety.

**Comment by bchawke** [comment]
> When I voiced my fears about NWAF _returning to the state it had been in 1.8.6 and earlier patches_ in #833 (which prompted the gutting of the barracks loot table to begin with)...

Wrong.  1.8.6 barracks spawned these weapons:

- {"M249_DZ","weapon",0.02},
- {"M24_DZ","weapon",0.01},
- {"M240_DZ","weapon",0.01},
- {"Mk48_DZ","weapon",0.01},
- {"M40A3_DZ","weapon",0.01},

Look at the code changes in #833.  They aren't there.  My suggestion does not return to 1.8.6.  Please stop saying that.

> You made it clear that you did not agree that there ever had been a problem but rather hailed it as "one of the best design features of DayZ Mod."

I said "The _focus on NWAF_ is not an 'age old problem'."  That doesn't mean that I think 1.8.6 and earlier loot tables were fine.  Let me use the term again: _middle ground_.  My proposal brings value back to NWAF, but does not take loot back to 1.8.6 values which I agree were OP.

> making hilarious suggestions...

This rhetoric isn't helpful at all

> ...like effectively cutting the amount of crashes in half based upon pictures you have seen of somebody's test server that's only restarted manually or from some none vanilla server which is hardly relevant on this github.

You've completely missed the point yet again.  Not going to re-explain it.

> I don't know how one is supposed to draw any other conclusion from your voiced opinions and suggested changes that you wish to have focus return to the NWAF and diminish the viability and lure of roaming for crashsites.

Did you miss the part where I suggested making heli crashes more rewarding by raising the chances of top tier loot and removing low tier military loot?  Also the part where I said I'm undecided as to whether more heli crashes with lower tier loot is better/worse than fewer with higher tier loot?  Scroll up, and please read my comments in context.

> As for justifying these changes by claiming they will narrow the "enormous" gap between new players/lone wolfs and geared squads I don't see how that's supposed to happen. Generally people who on top of spending a lot of time playing the game also group up will always wreck people who don't. Gear has little impact on this relationship. Also with your changes to the barracks lone wolfers and new players will have a harder time looting airfields than before meanwhile geared squads will loot cycle at will further bloating the loot economy in favor of geared players.

Wrong again.  I've already addressed this in #713 

Streatman, I disagree, as do many of the players I've talked to.  First off, on the server I play on there's no bases at NWAF because there's no point in building one there.  Bases are built around Stary and coastal cities or in remote locations out in the woods.  Second, even when NWAF is highly contested, it's possible to get in, loot, and get out, especially at night and when population is lower.  Third, NEAF has a barracks building which is a place that a lone wolf player can gear up since squads tend towards roaming the NW.  I don't have to convince you, though.  It's very evident from the state of the game right now.  Pre-1.8.7 it was possible to gear up by carefully looting the area and have a chance to defend yourself.  Post 1.8.7, lone wolves are running around with nothing but assault rifles while being taken out by squads that have every top tier weapon in the game.  Odds of finding a heli crash by yourself that hasn't been looted and actually has top tier guns is extremely low, much lower than the odds of making a successful barracks run back in 1.8.6.  There will always be a gap between lone wolves and squads, which is fine, it's a good mechanic to encourage teamwork.  The problem is the size of that gap.  It's too big now.

You keep bringing up base building.  Base building has several problems that need to be addressed.  Removing all the high end loot from the barracks is not the right way to solve those problems.

To be honest, I'm finished discussing this topic.  Literally anything I say (even when it's "heli crashes should have _**higher chances of spawning top tier loot**_") is brigaded by a vocal minority that takes my comments out of context to shoot down my ideas because they disagree with my opinions of barracks loot.  Carry on.

**Comment by Streatman** [comment]
"(even when it's "heli crashes should have higher chances of spawning top tier loot")"
thats the problem to begin with, dayz is a survival game and high tier stuff should be rare, rare for squads and lonewolves.
u think barracks need to change and i am okay with that, but i dont think that this is the way we should go.

i once suggested something. there are several "twin-guns", like M24/M40A3 and the L110/M249 when u just put one of each from the crashsides to the barracks that would be "only" 2 guns and a more suptile change then the massive move from crashsides to barracks.... all we say is that your suggestion is way over the top

IntenZ also has no bases on NWAF but thats due to the fakt that camping in a base without 3rd person perspective is impossible. but i can read the sidechat of europa discord and see ppl regularly that got killed from bases like that

**Comment by thcNox** [comment]
> Wrong. 1.8.6 barracks spawned these weapons:

> {"M249_DZ","weapon",0.02},
{"M24_DZ","weapon",0.01},
{"M240_DZ","weapon",0.01},
{"Mk48_DZ","weapon",0.01},
{"M40A3_DZ","weapon",0.01},

I've never said you wished to match the 1.8.6 loot table exactly so not exactly sure what you are on about here, grasping at straws? My claim have only been that you have bloated the barracks loot table to a point where the balance tips in favor of the NWAF effectively restoring NWAF to the state it had in 1.8.6.

> You've completely missed the point yet again. Not going to re-explain it.

You literally have this as one of four bullet points in  the thread starter:

> Set a maximum number of crash sites that can be present on the map. Three, maybe? We've seen the hilarious images of servers that have not been reset having dozens of crash sites littered about.

Less crash sites with higher chance of good loot is obviously going to favor the guy(s) with a chopper that you are so concerned about? You aren't consistent with your suggestions. One suggestion works against another suggested before and so on.

Zombies are on a tight performance budget as it is even with only 3 crash sites you'll have about 10% of the "zombie budget" tied up in crash site zombies. Also I'd argue that the effect would probably be marginal at best.

You keep chiding me for my rhetoric but at the same time you keep reminding us all that everyone you speak to and a large majority of players in general agree with your points whilst belittling everyone who oppose you by calling them a vocal minority as if to make your arguments somehow more valid and compelling. But you just come across as being childish instead.

**Comment by R4Z0R49** [comment]
ok so lets deal with this issue and the 4 points raised.

1, we wont be changing how they spawn, this was changed with the last big debate. If server admins are not happy with the current system they can change it to suit there player base/server.

2, Our system is timed around a 3 hour restart system if admins choose to run different to this system they will have to adjust the system to fit there own server.

3, Loot tables are already in discussion in another topic.

4, perm zeds wont happen, dayz zeds are spawned by players for a very good reason someone has to be able to control the movements. Moving this to the server will cause alot of issues the only way would be using a 2nd server and having that connect to the first to run all the AI (like SA).

In short if server admins are not happy with the current setup they are welcome to change it but we wont be changing from the system we use now,

Thanks
